refactor/urls.js

```diff
@@ -1,16 +1,37 @@
-// Refactor this code
+function getURL(dev = false, tasks = {}, additionalParams = {}) {
+  const BASE_URL = "/tasks";
+  const DEV_PARAMS = {
+    status: "ACTIVE",
+    dev: "true",
+    size: "20",
+  };
 
-function getURL(dev = false, tasks = {}) {
-    let url = dev
-        ? `/tasks?status=ACTIVE&dev=true&size=20`
-        : '/tasks';
+  tasks = tasks || {};
 
-    if (tasks.nextTasks) {
-        url += '?hasNext=true';
-    }
+  const params = new URLSearchParams();
 
-    if (tasks.prevTasks) {
-        url = '/tasks?hasPrev=true';
-    }
-    return { url };
+  if (dev) {
+    Object?.entries(DEV_PARAMS)?.forEach(([key, value]) => {
+      params?.append(key, value);
+    });
+  }
+
+  if (tasks?.nextTasks) {
+    params?.append("hasNext", "true");
+  }
+
+  if (tasks?.prevTasks) {
+    params?.delete("hasNext");
+    params?.append("hasPrev", "true");
+  }
+
+  Object?.entries(additionalParams)?.forEach(([key, value]) => {
+    params?.append(key, value);
+  });
+
+  const url = params?.toString()
+    ? `${BASE_URL}?${params?.toString()}`
+    : BASE_URL;
```

Comment:
Risky change, can this be handled more safely?

Reply:
tried adding